src/macros.rs

```diff
@@ -84,3 +84,99 @@ macro_rules! format_err {
         $crate::private::new_adhoc(format!($fmt, $($arg)*))
     };
 }
+
+/// Return early with an error and a status code.
+#[doc(hidden)]
+#[macro_export]
+macro_rules! bail2 {
```

Comment:
Maybe we should consider calling these new versions `{}_status`?

Reply:
Sure thing! Though they're only internal for now which means the naming doesn't matter too much -- for the next major we should probably make these the default as per https://github.com/http-rs/http-types/issues/227